### PR TITLE
Change the default X-Frame-Options header to be SAMEORIGIN

### PR DESCRIPTION
**Pull request by dvntucker**
Set the default X-Frame-Options header to be SAMEORIGIN

### CODE REVIEW

File: doc/en/user/source/production/config.rst [hunk 1]
```diff
@@ -74,3 +74,18 @@ In some circumstances, you might want to completely disable the web administrati
 * Set the Java system property GEOSERVER_CONSOLE_DISABLED to true by adding -DGEOSERVER_CONSOLE_DISABLED=true to your container's JVM options
 * Remove all of the gs-web*-.jar files from WEB-INF/lib
 
+X-Frame-Options Policy
+----------------------
+
+In order to prevent clickjacking attacks GeoServer defaults to setting the X-Frame-Options HTTP 
+header to DENY. This prevents GeoServer from being embedded into an iFrame, which prevents certain
```
Comment (jodygarnett) [inline doc/en/user/source/production/config.rst]: DENY or SAMEORIGIN

File: doc/en/user/source/production/config.rst [hunk 2]
```diff
@@ -74,3 +74,18 @@ In some circumstances, you might want to completely disable the web administrati
...
+kinds of security vulnerabilities. See the `OWASP Clickjacking entry <https://www.owasp.org/index.php/Clickjacking_Defense_Cheat_Sheet>`_ for details.
+
+If you wish to change this behavior you can do so through the following properties:
+
+* geoserver.xframe.shouldSetPolicy: controls whether the X-Frame-Options filter should be set at all. Default is true.
+* geoserver.xframe.policy: controls what the set the X-Frame-Options header to. Default is SAMEORIGIN valid options are DENY, SAMEORIGIN and ALLOW-FROM [uri]
+
+These properties can be set either via Java system property, command line argument (-D), environment
+variable or web.xml init parameter.
```
Comment (jodygarnett) [inline doc/en/user/source/production/config.rst]: Looks great, thanks

File: src/main/src/main/java/org/geoserver/filters/XFrameOptionsFilter.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+ * These properties can be set via command line -D arg, web.xml init or environment variable.
+ */
+public class XFrameOptionsFilter implements Filter {
+
+    private static final boolean DEFAULT_SHOULD_SET_POLICY = true;
+    private static final String DEFAULT_FRAME_POLICY = "SAMEORIGIN";
+    private static final String X_FRAME_OPTIONS = "X-Frame-Options";
+    public static final String GEOSERVER_XFRAME_SHOULD_SET_POLICY =
+        "geoserver.xframe.shouldSetPolicy";
+    public static final String GEOSERVER_XFRAME_POLICY = "geoserver.xframe.policy";
```
Comment (jodygarnett) [inline src/main/src/main/java/org/geoserver/filters/XFrameOptionsFilter.java]: I usually javadoc public fields

**Comment by aaime** [comment]
Ah, just saw this one, yep, better than DENY, sorry for the noise on the other pull.

**Comment by dvntucker** [comment]
Fixed the doc typo.

**Comment by dvntucker** [comment]
Ok, this should be ready to go.

**Pull request merged**